tests/migration/online_migration/yast2_migration.pm

```diff
@@ -209,7 +209,12 @@ sub run {
     send_key "alt-p";    # focus on the item of possible migration targets
     assert_screen 'yast2-migration-target-list-selected', 60;
     send_key_until_needlematch 'migration-target-' . get_var("VERSION"), 'down', 20, 3;
-    send_key "alt-n";
+
+    # On some platforms such as Aarch64 and x86_64, after sending alt-n keys,
+    # we need to wait a while for the next screen to show up.
+    send_key("alt-n", wait_screen_change => 1);
+    wait_still_screen 3;
```

Comment:
You should not need to do `wait_screen_change+wait_still_screen+assert_screen` . In line https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12499/files#diff-283b909b6e798ec5f92a947c75b4e9ef2cc6805d36e0cdcfadab2b8fbd0a80a0R221 as well as https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12499/files#diff-283b909b6e798ec5f92a947c75b4e9ef2cc6805d36e0cdcfadab2b8fbd0a80a0R228 there are `assert_screen` calls as next calls so this should already ensure that the next screen shows up as expected. How should `wait_screen_change` or `wait_still_screen` help?

Reply:
removed 'wait_still_screen 3' and verified that it passed. https://openqa.nue.suse.com/tests/5973295, thanks

Reply:
Well, you have it replaced by sleep 3, that's not better.

You are using `sleep()` or are simulating a sleep-like functionality. This is introducing the risk of either unnecessarily wasting time or just shifting race conditions and making them even harder to investigate. You *can* use sleep but there are only very limited cases where is the best solution (e.g. yield threads in low-level system code). Here you have the classical problem in software design of [Synchronization](https://en.wikipedia.org/wiki/Synchronization_(computer_science)) . sleep should not be used in place of proper synchronization methods. https://blogs.msmvps.com/peterritchie/2007/04/26/thread-sleep-is-a-sign-of-a-poorly-designed-program/ is one of multiple references that explains this: "The original problem is likely a timing/synchronization issue, ignoring it by hiding it with Thread.Sleep is only going to delay the problem and make it occur in random, hard to reproduce ways." And if the problem does not occur at least time is wasted. https://testing.googleblog.com/2021/03/test-flakiness-one-of-main-challenges.html states that as well: "Do NOT add arbitrary delays as these can become flaky again over time and slow down the test unnecessarily."

We are waiting because some condition changes some time ... keyword(s) is/are *some time*. It is important to know the exact condition that the code should wait for and check for that instead of delaying program execution for an arbitrary amount of time. For openQA tests in general just try to do what you as a human user would also do. You would not look on your watch waiting for exactly 100s before you look at the screen again, right? :)

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details

Reply:
just found this case, It needs to send 'alt-n' more than 1 times in the dev mode. it's just a try and will remove it. thanks.
./hj-branch.sh -j 5896259 -b send_next -s "_GROUP=0"
 https://openqa.nue.suse.com/t5973259
https://openqa.nue.suse.com/t5973429

Reply:
OK, thanks a lot.